### DISCUSSION TITLE
Does the plugin wait for the hotend to cool before turning off? and Ok to update firmware?

### DISCUSSION THREAD

**Discussion by PICKLEBOY** [jamesmccannon02/OctoPrint-Tplinkautoshutdown#72] [Q&A]
I haven't used it yet, currently waiting for my HS105s to ship, but does the auto off feature wait for the hotend to cool off before shutting the printer down or does it simply turn off right when the print is finished?

Also does the smart plug have to be used locally only or can I update its firmware? The older TPlink plugin for octoprint stated later firmwares caused problems for that plugin.

**Comment by CodingTheUnknown** [comment]
Hi this does not wait for the hot end to cool but personally never had an issue with it. In terms of hardware updates it's tricky and somewhat trial and error

**Reply by PICKLEBOY** [reply]
Ok, thanks for the quick reply. I’ve just read that it’s best to wait for the hotend to cool a little bit before turning off to avoid potential heat creep is why I asked. I assume once installed there isn’t a way to enable a delay myself for the autoshutoff in my slicer gcode?
